cmd/bootstrap/cmd/partner_infos.go

```diff
@@ -0,0 +1,244 @@
+	// fields not needed are left out.
+	idField = iota
+	roleField
+	networkingAddressField
+	networkingKeyField
+	stakingKeyField
+	tokensStakedField
+	nodeInfosPubDir       = "node-pub-infos"
+	infoFileNameTemplate  = "node-info.pub.%s.json"
+	partnerStakesFileName = "partner-stakes.json"
+	flowNodeAddrPart      = "nodes.onflow.org"
+	defaultPartnerStake   = "100"
+)
+
+// NodePubInfo basic representation of node-pub-info.json data
+type NodePubInfo struct {
+	Role          string `json:"Role"`
+	Address       string `json:"Address"`
+	NodeID        string `json:"NodeID"`
+	Stake         string `json:"Stake"`
+	NetworkPubKey string `json:"NetworkPubKey"`
+	StakingPubKey string `json:"StakingPubKey"`
+}
+
+// PartnerStakesInfo mapping of NodeID =>
+type PartnerStakesInfo map[string]string
+
+// populatePartnerInfos represents the `populate-partner-infos` command which will read the proposed node
+// table from the staking contract and for each identity in the proposed table generate a node-info-pub
+// json file. It will also generate the partner-stakes json file.
+var populatePartnerInfosCMD = &cobra.Command{
+	Use:   "populate-partner-infos",
+	Short: "Generates a node-info-pub-*.json for all proposed identities in staking contract and corresponding partner-stakes.json file.",
+	Run:   populatePartnerInfosRun,
+}
+
+func init() {
+	rootCmd.AddCommand(populatePartnerInfosCMD)
+
+	populatePartnerInfosCMD.Flags().StringVar(&flagOutputDir, "out", "", "the directory where the generated node-info-pub files will be written")
+	populatePartnerInfosCMD.Flags().StringVar(&flagANAddress, "access-address", "", "the address of the access node used for client connections")
+	populatePartnerInfosCMD.Flags().StringVar(&flagANNetworkKey, "access-network-key", "", "the network key of the access node used for client connections in hex string format")
+	populatePartnerInfosCMD.Flags().StringVar(&flagNetworkEnv, "network", "", "the network string, expecting one of ( mainnet | testnet | localnet )")
+
+	cmd.MarkFlagRequired(populatePartnerInfosCMD, "out")
+	cmd.MarkFlagRequired(populatePartnerInfosCMD, "access-address")
+}
+
+// populatePartnerInfosRun generate node-pub-info file for each node in the proposed table and the partner stakes file, and prints the
+// address and node ID of any flow nodes that were skipped.
+func populatePartnerInfosRun(_ *cobra.Command, _ []string) {
+	ctx := context.Background()
+
+	env, err := common.EnvFromNetwork(flagNetworkEnv)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get environment for network (%s)", flagNetworkEnv)
+	}
+
+	flowClient := getFlowClient()
+
+	proposedNodeIDS, err := executeGetProposedTableScript(ctx, env, flowClient)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get proposed table")
+	}
+
+	partnerStakes := make(PartnerStakesInfo)
+	skippedNodes := 0
+	numOfNodesByType := map[string]int{
```

Comment:
```suggestion
	numOfPartnerNodesByRole := map[string]int{
```
Better to be clear this is only counting partner nodes

Nitpick: role terminology

Reply:
https://github.com/onflow/flow-go/pull/1988/commits/c34743b2afef68f94ebeeae98cdca9695a565740